File: leetcode/1001-2000/1701-1800/1731-1740/1732-find-the-highest-altitude/FindTheHighestAltitude.ts
function largestAltitude(gain: number[]): number {
  const { max } = gain.reduce(
    (accumulated, currentValue) => {
      let { max, current } = accumulated;

      current = current + currentValue;
      max = Math.max(max, current);

      return { max, current };
    },
    {
      max: 0,
      current: 0
    }
  );

  return max;
};

console.log(largestAltitude([-5, 1, 5, 0, -7]));

File: leetcode/1001-2000/1701-1800/1731-1740/1732-find-the-highest-altitude/FindTheHighestAltitude.java
public class FindTheHighestAltitude {
  public static void main(String[] args) {
    int[] gain = new int[] {
        -5, 1, 5, 0, -7
    };

    System.out.println(new Solution().largestAltitude(gain));
  }
}

class Solution {
  public int largestAltitude(int[] gain) {
    int max = 0;
    int current = 0;

    for (int i = 0; i < gain.length; i++) {
      current += gain[i];

      max = Math.max(max, current);
    }

    return max;
  }
}
